pennylane/fourier/advanced_spectrum.py

```diff
@@ -0,0 +1,365 @@
+        That is, a parameter that does not control any gate has the spectrum ``[0]``.
+
+    Args:
+        qnode (pennylane.QNode): :class:`~.pennylane.QNode` to compute the spectrum for
+        encoding_args (dict[str, list[tuple]], set): Parameter index dictionary;
+            keys are argument names, values are index tuples for that argument
+            or an ``Ellipsis``. If a ``set``, all values are set to ``Ellipsis``.
+            The contained argument and parameter indices indicate the scalar variables
+            for which the spectrum is computed
+        argnum (list[int]): Numerical indices for arguments with respect to which
+            to compute the spectrum
+        decimals (int): number of decimals to which to round frequencies.
+        num_pos (int): Number of additional random positions at which to evaluate the
+            Jacobian of the preprocessing and test that it is constant.
+            Setting ``num_pos=0`` will deactivate the test.
+
+    Returns:
+        function: Function which accepts the same arguments as the QNode.
+            When called, this function will return a dictionary of dictionaries
+            containing the frequency spectra per QNode parameter.
+
+    **Details**
+
+    A circuit that returns an expectation value of a Hermitian observable which depends on
+    :math:`N` scalar inputs :math:`x_j` can be interpreted as a function
+    :math:`f: \mathbb{R}^N \rightarrow \mathbb{R}` (as the observable is Hermitian,
+    the expectation value is real-valued).
+    This function can always be expressed by a Fourier-type sum
+
+    .. math::
+
+        \sum \limits_{\omega_1\in \Omega_1} \dots \sum \limits_{\omega_N \in \Omega_N}
+        c_{\omega_1,\dots, \omega_N} e^{-i x_1 \omega_1} \dots e^{-i x_N \omega_N}
+
+    over the *frequency spectra* :math:`\Omega_j \subseteq \mathbb{R},`
+    :math:`j=1,\dots,N`. Each spectrum has the property that
+    :math:`0 \in \Omega_j`, and the spectrum is symmetric
+    (i.e., for every :math:`\omega \in \Omega_j` we have that :math:`-\omega \in\Omega_j`).
+    If all frequencies are integer-valued, the Fourier sum becomes a *Fourier series*.
+
+    As shown in `Vidal and Theis (2019) <https://arxiv.org/abs/1901.11434>`_ and
+    `Schuld, Sweke and Meyer (2020) <https://arxiv.org/abs/2008.08605>`_,
+    if an input :math:`x_j, j = 1 \dots N`,
+    only enters into single-parameter gates of the form :math:`e^{-i x_j G}`
+    (where :math:`G` is a Hermitian generator),
+    the frequency spectrum :math:`\Omega_j` is fully determined by the eigenvalues
+    of the generators :math:`G`. In many situations, the spectra are limited
+    to a few frequencies only, which in turn limits the function class that the circuit
+    can express.
+
+    The ``advanced_spectrum`` function computes all frequencies that will
+    potentially appear in the sets :math:`\Omega_1` to :math:`\Omega_N`.
```

Comment:
great explanation!

Reply:
Agreed, courtesy of Olivia and Maria! :)